### PR TITLE
[DEVEX-158] Improve pre-commit hook

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Update pre-commit-terraform to 1.94.1
Fixed an issue where if multiple extensions were passed to `--files` argument, pre-commit skipped all steps for some reason. Now only `tf` files are passed
Added suggested step to "fix" a dependency issue of the base image container
Run pre-commit over everything if there are no `tf` files changed
Differentiate the behavior of `terraform_docs` for modules and resources, whether including or not provider info
`terraform_docs` creates `README.md` files if they don't exist
Introduced `terraform_trivy` with its `.trivyignore` file

Fixed README.md according to new rules
Fixed lock files by adding missing SHAs

### Motivation and context

Previous implementation had some issue while verifying files at PR creation and dependencies were quite old. Security checks were missing as well.

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### CODE REVIEW

File: .github/workflows/static_analysis.yaml [hunk 1]
```diff
@@ -20,7 +25,7 @@ jobs:
     env:
       TERRAFORM_VERSION: "1.7.5"
     container:
-      image: ghcr.io/antonbabenko/pre-commit-terraform:v1.86.0@sha256:a1ffb8fff155934c1937580eeab8b3f68726485cd54025d2f911b0c21a9becba
+      image: ghcr.io/antonbabenko/pre-commit-terraform:v1.94.1
```
Comment (gunzip) [inline .github/workflows/static_analysis.yaml:28]: what about version pinning here?

File: .github/workflows/static_analysis.yaml [hunk 2]
```diff
@@ -47,18 +52,18 @@ jobs:
...
+          DIFF="$(git diff --name-only origin/${{ github.base_ref }} origin/${{ github.head_ref }} -- '*.tf')"
+          echo "Diff between ${{ github.base_ref }} and ${{ github.head_ref }}"
           echo "files=$(echo \"$DIFF\" | xargs echo)" >> $GITHUB_OUTPUT
 
-      - name: Cache pre-commit
-        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
-        with:
-          path: ~/.cache/pre-commit
-          key: pre-commit-3|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Fix Alpine Container Image
```
Comment (gunzip) [inline .github/workflows/static_analysis.yaml:59]: why it's not cached anymore?

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
-    rev: v1.86.0
+    rev: v1.94.1
     hooks:
       - id: terraform_fmt
       - id: terraform_docs
+        name: terraform_docs on modules
+        args:
+          - --hook-config=--create-file-if-not-exist=true
+          - --args=--hide providers
+        exclude: |
```
Comment (gunzip) [inline .pre-commit-config.yaml:11]: can we switch to `include infra/modules` pattern instead?

File: .trivyignore [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# Github repository shouldn’t be public.
```
Comment (gunzip) [inline .trivyignore:1]: can you add a link with the trivy related docs?

File: .github/workflows/static_analysis.yaml (hunk 2 above)
Comment (Krusty93) [reply .github/workflows/static_analysis.yaml:59]: I didn't notice any performance boost, it seemed quite useless

File: .github/workflows/static_analysis.yaml (hunk 1 above)
Comment (Krusty93) [reply .github/workflows/static_analysis.yaml]: Added

File: .trivyignore (hunk 4 above)
Comment (Krusty93) [reply .trivyignore:1]: Do you mean for each suppressed warning or trivy in general?

File: .pre-commit-config.yaml (hunk 3 above)
Comment (Krusty93) [reply .pre-commit-config.yaml:11]: probably using `[files](https://pre-commit.com/#top_level-files)`, but I have to give it a try

Comment (mamu0) [reply .pre-commit-config.yaml:11]: Yes, if we want to specify that the only folder to check is modules, we can use `files: ^infra/modules/.*`, which will automatically exclude everything else

File: .trivyignore (hunk 4 above)
Comment (gunzip) [reply .trivyignore:1]: in general (with a list of possible configuration)

Comment (Krusty93) [reply .trivyignore:3]: Updated with the main page of misconfiguration docs

**Review by gunzip — APPROVED** [review]

**Review by christian-calabrese — APPROVED** [review]

**Pull request merged**